src/aws-fsx/maintenance-time.ts

```diff
@@ -0,0 +1,74 @@
+import { aws_fsx } from 'aws-cdk-lib';
+
+/**
+ * Properties required for setting up a weekly maintenance time
+ */
+export interface MaintenanceTimeProps {
+  /**
+   * The day of the week for maintenance to be performed.
+   */
+  readonly day: aws_fsx.Weekday;
+  /**
+   * The hour of the day (from 0-23) for maintenance to be performed.
+   */
+  readonly hour: number;
+  /**
+   * The minute of the hour (from 0-59) for maintenance to be performed.
+   */
+  readonly minute: number;
+}
+
+/**
+ * Class for scheduling a weekly maintenance time.
+ */
+export class MaintenanceTime {
```

Comment:
Is this class for use outside of ONTAP? Or is it ONTAP specific?

Reply:
I think this class is used for more than just ONTAP.

There is a LustreMaintenanceTime class in aws-cdk-lib that is similar to the MaintenanceTime class. I've added the MaintenanceTime class because the LustreMaintenanceTime class appears to be specific to the Lustre file system. By using the MaintenanceTime class, we can apply it to other FSx file system constructs as well.